### PR TITLE
Update k8s-infra-prow-build README to clarify access instructions.

### PR DESCRIPTION
**Pull request by cjwagner**
The instructions for accessing the cluster were a bit hidden in the setup instructions. I pulled them out into their own section for clarity (and to link to more easily) and also fixed a minor mistake and an omission.

/assign @spiffxp

### CODE REVIEW

File: infra/gcp/clusters/projects/k8s-infra-prow-build/prow-build/README.md [hunk 1]
```diff
@@ -4,6 +4,25 @@ These terraform resources define a GCP project containing a GKE cluster
...
+## Accessing the control plane
+
+The k8s-infra-prow-build cluster is a [private GKE cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters).
+It cannot be directly accessed with kubectl from outside the GCP project,
+cloud-shell must first be used as follows:
+```shell
+# Start a cloud shell
+gcloud alpha cloud-shell ssh --project=k8s-infra-prow-build
+
+# Login to set the authenticated user for gcloud in the cloud shell session.
```
Comment (spiffxp) [inline infra/gcp/clusters/projects/k8s-infra-prow-build/prow-build/README.md:16]: As of https://github.com/kubernetes/k8s.io/pull/1727 this part is no longer true, so here's my suggestion to make this more relevant.  Can move the bookmarks to the bottom of the page.
```suggestion
Access to the k8s-infra-prow-build project hosting the cluster is granted by membership in one of two @kubernetes.io groups:
- [k8s-infra-prow-oncall@kubernetes.io][k8s-infra-prow-oncall@]: grants [`roles/owner`][roles/owner] access
- [k8s-infra-prow-viewers@kubernetes.io][k8s-infra-prow-viewers@]: grants [`prow.viewer`][roles/prow.viewer] access

If you are not a member of either of these groups, please [follow these instructions to join][join-groups]

[k8s-infra-prow-oncall]: https://github.com/kubernetes/k8s.io/blob/3a1aea1652f02a95253402bde2bca63cb4292f8e/groups/groups.yaml#L647-L670
[k8s-infra-prow-viewers]: https://github.com/kubernetes/k8s.io/blob/3a1aea1652f02a95253402bde2bca63cb4292f8e/groups/groups.yaml#L672-L699
[roles/owner]: https://cloud.google.com/iam/docs/understanding-roles#basic-definitions
[roles/prow.viewer]: https://github.com/kubernetes/k8s.io/blob/main/infra/gcp/roles/prow.viewer.yaml
[join-groups]: https://github.com/kubernetes/k8s.io/tree/main/groups#making-changes

```shell
# Login to set the authenticated user for gcloud
```

**Review by spiffxp — CHANGES_REQUESTED** [review]
So actually...

File: infra/gcp/clusters/projects/k8s-infra-prow-build/prow-build/README.md (hunk 1 above)
Comment (cjwagner) [reply infra/gcp/clusters/projects/k8s-infra-prow-build/prow-build/README.md]: Awesome, added these changes.

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Pull request merged**